archive/imo/imo1969_q1.lean

```diff
@@ -0,0 +1,81 @@
+import tactic
+import tactic.basic
+import tactic.linarith
+import tactic.norm_cast
+import tactic.ring
+
+open int
+open nat
+
+/-
+The 1969 IMO, problem 1:
+Prove that there are infinitely many natural numbers $a$ with the following property:
+the number $z = n^4 + a$ is not prime for any natural number $n$.
+
+The key to the solution is that you can factor this into the product of two polynomials, if a = 4*m^4.
+-/
+
+lemma factorization (m n: ℤ): (n^2 + 2*m^2 - 2*n*m) * (n^2 + 2*m^2 + 2*n*m) = n^4 + 4*m^4
+:= by ring
+
+/-
+To show that the product is not prime, we need to show each of the factors is at least 2, which we can
+do with a sum-of-squares expression.
+-/
+lemma left_factor_large (m n: ℤ) (h: m > 1): (n^2 + 2*m^2 - 2*n*m) > 1 :=
```

Comment:
Mathlib convention is `h : 1 < m` and to avoid `>` in general.

Reply:
@lacker the reason behind this, is that currently our automation (like `simp`) get's confused when we mix `<` and `>`. It's really sad, but otoh, the automation is good enough that we've decided to avoid `>` to make it happier.
Since there will not be other files depending on this one, it is up to you whether you use `>` in your conclusions.
However, for example `(h : m > 1)` will not be as powerful to the automation as the same hypothesis stated with `<`.

Reply:
there is a certain irony to, I thought the idea of making a computer system powerful enough to solve IMO problems was exciting, and wanted to contribute, and now in the process I am discovering that the system sometimes has trouble remembering that a < b means the same thing as b > a.

anyway, I fixed everything to use < instead of >